### PR TITLE
Feature/support sort order in identities

### PR DESCRIPTION
**Pull request by sherlockvn**

### CODE REVIEW

File: lib/jmap/core/capability/capability_identifier.dart [hunk 1]
```diff
@@ -9,6 +9,7 @@ class CapabilityIdentifier with EquatableMixin {
   static final jmapMdn = CapabilityIdentifier(Uri.parse('urn:ietf:params:jmap:mdn'));
   static final jmapQuota = CapabilityIdentifier(Uri.parse('urn:ietf:params:jmap:quota'));
   static final jmapTeamMailboxes = CapabilityIdentifier(Uri.parse('urn:apache:james:params:jmap:mail:shares'));
+  static final jmapSortOrder = CapabilityIdentifier(Uri.parse('urn:apache:james:params:jmap:mail:identity:sortorder'));
```
Comment (hoangdat) [inline lib/jmap/core/capability/capability_identifier.dart:12]: should be `jamesSortOrder`

File: pubspec.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ dev_dependencies:
   flutter_test:
     sdk: flutter
 
-  http_mock_adapter: 0.3.2
+  http_mock_adapter: 0.3.3
```
Comment (hoangdat) [inline pubspec.yaml:34]: why we need to upgrade it?

File: test/jmap/identities/create_identity_with_sort_order_test.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import 'package:jmap_dart_client/jmap/identities/set/set_identity_method.dart';
+import 'package:jmap_dart_client/jmap/identities/set/set_identity_response.dart';
+import 'package:jmap_dart_client/jmap/jmap_request.dart';
+
+void main() {
+  group('test to json set identity method', () {
+    final expectedCreated = Identity(
+      id: IdentityId(Id('5ccf6d7b-77e8-467a-9064-9f7ccfb19e86')),
+      sortOrder: UnsignedInt(99999),
+
```
Comment (dab246) [inline test/jmap/identities/create_identity_with_sort_order_test.dart:18]: remove excess line

File: test/jmap/identities/create_identity_with_sort_order_test.dart [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    );
+
+    test('create new identity with sort order method and response parsing', () async {
+      final baseOption  = BaseOptions(method: 'POST');
+      final dio = Dio(baseOption)
+        ..options.baseUrl = 'https://gateway.upn.integration-open-paas.org';
```
Comment (dab246) [inline test/jmap/identities/create_identity_with_sort_order_test.dart:24]: The `server` tests should not be used here.

File: test/jmap/identities/get_identities_with_sort_order_test.dart [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        id: IdentityId(Id('f1e627f0-e21b-371a-9c03-ac74d06353fd')),
+        email: 'firstname37.surname37@upn.integration-open-paas.org',
+        sortOrder: UnsignedInt(1000),
+        bcc: {EmailAddress(null, "dpham@gmail.com")}
+    );
+
+    test('get identity method and response parsing', () async {
+      final baseOption  = BaseOptions(method: 'POST');
+      final dio = Dio(baseOption)
+        ..options.baseUrl = 'https://gateway.upn.integration-open-paas.org';
```
Comment (dab246) [inline test/jmap/identities/get_identities_with_sort_order_test.dart:34]: same

File: test/jmap/identities/create_identity_with_sort_order_test.dart [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+            htmlSignature: Signature('<body><div>Dat T. Vu <br>Mobile Engineer <br>LINAGORA VIETNAM <br>A: 8th Floor (Toong VPBank Tower, No. 5 Dien Bien Phu  Str., Ba Dinh Dist., Ha Noi <br>P: (+84) 366-769-439<br>E: tdvu@linagora.com</div></body>'),
+            sortOrder: UnsignedInt(99999),
+          )
+        );
+
+      final httpClient = HttpClient(dio);
+      final requestBuilder = JmapRequestBuilder(httpClient, ProcessingInvocation());
+      final setIdentityInvocation = requestBuilder.invocation(setIdentityMethod);
+      final response = await (requestBuilder
+          ..usings(setIdentityMethod.requiredCapabilities))
```
Comment (dab246) [inline test/jmap/identities/create_identity_with_sort_order_test.dart:92]: why you can not use `capability` of `sortOrder?

File: test/jmap/identities/get_identities_with_sort_order_test.dart [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/core/unsigned_int.dart';
+import 'package:jmap_dart_client/jmap/identities/get/get_identity_method.dart';
+import 'package:jmap_dart_client/jmap/identities/get/get_identity_response.dart';
+import 'package:jmap_dart_client/jmap/identities/identity.dart';
+import 'package:jmap_dart_client/jmap/jmap_request.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+
+void main() {
+  group('test to json get identity method', () {
```
Comment (hoangdat) [inline test/jmap/identities/get_identities_with_sort_order_test.dart:16]: @sherlockvn please add the test for the case:
- increase `sortOrder` of default Identity 
- create new default Identity 
All in one request

**Review by hoangdat — APPROVED** [review]

**Pull request merged**